src/instana/instrumentation/psycopg2.py

```diff
@@ -5,33 +5,44 @@
 import copy
 import wrapt
 
-from ..log import logger
-from .pep0249 import ConnectionFactory
+from typing import Callable, Optional, Any, Tuple, Dict
+from instana.log import logger
+from instana.instrumentation.pep0249 import ConnectionFactory
 
 try:
     import psycopg2
     import psycopg2.extras
 
-    cf = ConnectionFactory(connect_func=psycopg2.connect, module_name='postgres')
+    cf = ConnectionFactory(connect_func=psycopg2.connect, module_name="postgres")
 
-    setattr(psycopg2, 'connect', cf)
-    if hasattr(psycopg2, 'Connect'):
-        setattr(psycopg2, 'Connect', cf)
+    setattr(psycopg2, "connect", cf)
+    if hasattr(psycopg2, "Connect"):
+        setattr(psycopg2, "Connect", cf)
 
-    @wrapt.patch_function_wrapper('psycopg2.extensions', 'register_type')
-    def register_type_with_instana(wrapped, instance, args, kwargs):
+    @wrapt.patch_function_wrapper("psycopg2.extensions", "register_type")
+    def register_type_with_instana(
+        wrapped: Callable[..., Any],
+        instance: Optional[Any],
+        args: Tuple[Any, ...],
+        kwargs: Dict[str, Any],
+    ) -> Callable[..., Any]:
         args_clone = list(copy.copy(args))
 
-        if (len(args_clone) >= 2) and hasattr(args_clone[1], '__wrapped__'):
+        if (len(args_clone) >= 2) and hasattr(args_clone[1], "__wrapped__"):
             args_clone[1] = args_clone[1].__wrapped__
 
         return wrapped(*args_clone, **kwargs)
 
-    @wrapt.patch_function_wrapper('psycopg2._json', 'register_json')
-    def register_json_with_instana(wrapped, instance, args, kwargs):
-        if 'conn_or_curs' in kwargs:
-            if hasattr(kwargs['conn_or_curs'], '__wrapped__'):
-                kwargs['conn_or_curs'] = kwargs['conn_or_curs'].__wrapped__
+    @wrapt.patch_function_wrapper("psycopg2._json", "register_json")
+    def register_json_with_instana(
+        wrapped: Callable[..., Any],
+        instance: Optional[Any],
+        args: Tuple[Any, ...],
+        kwargs: Dict[str, Any],
+    ) -> Callable[..., Any]:
+        if "conn_or_curs" in kwargs:
+            if hasattr(kwargs["conn_or_curs"], "__wrapped__"):
+                kwargs["conn_or_curs"] = kwargs["conn_or_curs"].__wrapped__
```

Comment:
Same as [this](https://github.com/instana/python-sensor/pull/597/files#r1734075703) comment.

Reply:
Done